lib/Driver/WindowsToolChains.cpp

```diff
@@ -193,3 +195,28 @@ toolchains::Windows::constructInvocation(const LinkJobAction &job,
 
   return II;
 }
+
+ToolChain::InvocationInfo
+toolchains::Windows::constructInvocation(const ArchiveJobAction &job,
+                                         const JobContext &context) const {
+   assert(context.Output.getPrimaryOutputType() == file_types::TY_Image &&
+         "Invalid linker output type.");
+
+  ArgStringList Arguments;
+
+  // Configure the toolchain.
+  const char *Lib = "lib";
```

Comment:
Please use `llvm-ar` if available, otherwise fall back to `lld-link` (`lld-link -lib`), otherwise try `lib`.  I don't think that we need to bother with `link /LIB` as a means of invoking the librarian (`lib`).

Reply:
That seems a lot of complexity – three tools with different invocations. I assume you want `llvm-ar` and `lld-link` for cross-compilation, so would it be enough to just pick one of those and say it's required for cross-compilation?

Reply:
I don't want to depend on `lld-link` yet.  Until we have swift-lld created and tracking, please do not depend on `lld-link`.  However, `lld-link` and `link` should be compatible enough, and `link -lib` is the same as `lib.exe`.  So if you want to just use link to invoke the librarian, I am fine with that for Windows.  You can use either `lld-link` or `link` then to invoke `lib`.